tests/test_integration.py

```diff
@@ -0,0 +1,113 @@
+
+    # cloning with `keep_solution=True` keeps all timeseries and the solution
+    # (same behaviour as `ixmp.Scenario`)
+    scen2 = scen.clone(keep_solution=True)
+    assert np.isclose(scen2.var('OBJ')['lvl'], 153.675)
+    pdt.assert_frame_equal(scen2.timeseries(iamc=True), TS_DF)
+
+    # cloning with `keep_solution=False` drops the solution and only keeps
+    # timeseries set as `meta=True` or prior to the first model year
+    # (DIFFERENT behaviour from `ixmp.Scenario`)
+    scen3 = scen.clone(keep_solution=False)
+    assert np.isnan(scen3.var('OBJ')['lvl'])
+    pdt.assert_frame_equal(scen3.timeseries(iamc=True), TS_DF_CLEARED)
+
+
+def test_run_remove_solution(test_mp):
+    # create a new instance of the transport problem and solve it
+    scen = make_dantzig(test_mp, solve=True)
+    assert np.isclose(scen.var('OBJ')['lvl'], 153.675)
+
+    # check that re-solving the model will raise an error if a solution exists
+    pytest.raises(ValueError, scen.solve)
+
+    # check that removing solution with a first-model-year arg raises an error
+    # (DIFFERENT behaviour from `ixmp.Scenario`)
+    pytest.raises(TypeError, scen.remove_solution, first_model_year=1964)
+
+    # check that removing solution does not delete timeseries data
+    # before first model year (DIFFERENT behaviour from `ixmp.Scenario`)
+    scen.remove_solution()
+    pdt.assert_frame_equal(scen.timeseries(iamc=True), TS_DF_CLEARED)
+
+
+def test_shift_first_model_year(test_mp):
+    scen = make_dantzig(test_mp, solve=True, multi_year=True)
+    clone = scen.clone(shift_first_model_year=1964)
+
+    # check that solution and timeseries in new model horizon have been removed
+    assert np.isnan(clone.var('OBJ')['lvl'])
+    pdt.assert_frame_equal(clone.timeseries(iamc=True), TS_DF_SHIFT)
+    # check that the variable `ACT` is now the parameter `historical_activity`
+    assert not clone.par('historical_activity').empty
+
+
+def scenario_list(mp):
+    return mp.scenario_list(default=False)[['model', 'scenario']]
+
+
+def assert_multi_db(mp1, mp2):
+    pdt.assert_frame_equal(scenario_list(mp1), scenario_list(mp2))
+
+
+def test_multi_db_run(tmpdir):
+    # create a new instance of the transport problem and solve it
+    mp1 = Platform(tmpdir / 'mp1', dbtype='HSQLDB')
+    scen1 = make_dantzig(mp1, solve=True)
+
+    mp2 = Platform(tmpdir / 'mp2', dbtype='HSQLDB')
+    # add other unit to make sure that the mapping is correct during clone
+    mp2.add_unit('wrong_unit')
+    mp2.add_region('wrong_region', 'country')
+
+    # check that cloning across platforms must copy the full solution
+    dest = dict(platform=mp2)
+    pytest.raises(ValueError, scen1.clone, **dest, keep_solution=False)
+    pytest.raises(ValueError, scen1.clone, **dest, shift_first_model_year=1964)
+
+    # clone solved model across platforms (with default settings), close the db
+    scen1.clone(platform=mp2, keep_solution=True)
+
+    mp2.close_db()
```

Comment:
maybe add a comment as to why these lines are critical so someone in the future doesn't take them out by accident

Reply:
done

Reply:
also made a related PR in https://github.com/iiasa/ixmp/pull/151 to have the same docstrings there